compiler/hash-source/src/location.rs

```diff
@@ -85,6 +85,21 @@ impl ByteRange {
     pub fn into_location(self, source_id: SourceId) -> Span {
         Span::new(self, source_id)
     }
+
+    /// Check if one span intersects with another one.
```

Comment:
maybe a better name is `touches` since `intersects` implies they can overlap

Reply:
Also the ordering matters, maybe a better name is actually `a.is_right_after(b)`?